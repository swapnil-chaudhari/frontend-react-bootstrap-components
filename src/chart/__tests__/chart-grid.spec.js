import React from 'react';
import renderShallow from 'render-shallow';
import ChartGrid from '../chart-grid';

describe('<ChartGrid>', () => {
    describe('when it renders', () => {
        const props = {
            xTicks: [1, 2],
            yTicks: [3, 4],
            xScale: (tick) => tick * 10,
            yScale: (tick) => tick * 100,
            xOffset: 5,
            width: 888,
            height: 400
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<ChartGrid { ...props } />).output;
        });

        test('has proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
