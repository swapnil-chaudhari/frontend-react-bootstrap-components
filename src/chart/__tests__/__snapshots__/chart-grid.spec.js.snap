// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartGrid> when it renders has proper markup 1`] = `
<g
  className="ChartGrid"
>
  <line
    className="ChartGrid-y-line"
    x1={5}
    x2={893}
    y1={300}
    y2={300}
  />
  <line
    className="ChartGrid-y-line"
    x1={5}
    x2={893}
    y1={400}
    y2={400}
  />
  <line
    className="ChartGrid-x-line"
    x1={15}
    x2={15}
    y2={400}
  />
  <line
    className="ChartGrid-x-line"
    x1={25}
    x2={25}
    y2={400}
  />
</g>
`;
